file-store/src/main/java/com/expedia/blobs/stores/io/FileStore.java

```diff
@@ -15,47 +15,53 @@
  *      limitations under the License.
  *
  */
-package com.expedia.blobs.core.io;
+package com.expedia.blobs.stores.io;
 
 import com.expedia.blobs.core.Blob;
 import com.expedia.blobs.core.BlobReadWriteException;
 import com.expedia.blobs.core.SimpleBlob;
+import com.expedia.blobs.core.io.AsyncSupport;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.reflect.TypeToken;
+
 import java.io.File;
 import java.io.IOException;
 import java.lang.reflect.Type;
 import java.nio.charset.Charset;
 import java.util.Map;
 import java.util.Optional;
+
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.FilenameUtils;
 import org.apache.commons.lang.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class FileStore extends AsyncSupport {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileStore.class);
     private final File directory;
     private final Gson gson = new GsonBuilder().create();
     private final Type mapType = new TypeToken<Map<String, String>>() {
     }.getType();
     private final Charset Utf8 = Charset.forName("UTF-8");
+    ShutdownHook shutdownHook;
```

Comment:
Do we need a separate class? it can be an inline lambda to close from a Runtime hook

Reply:
It could be an inline lambda but I needed the instance of the shutdown hook saved for testing purposes which is why I had to create a separate class for the shutdown hook. To test if a specific shutdown hook has been added to the AutoShutdownHooks I need the same instance.

Reply:
Can you check the tests? Because the class is a private one `private class ShutdownHook extends Thread`  - test is checking for presence of an instance not this instance

Reply:
So, I am creating a shutdown hook and saving it in the instance of the store to be tested in the test case if it exists. If the store is present then I will try to remove the AutoShutDownHook from JVM using the instance that was created by the store.
I couldn't find out a way to figure out if that particular instance was added as a hook or not. What I could do was to see if at least one instance is present in AutoShutDownHook to be removed.

Reply:
Do this
```
private Thread shutdownHook = new Thread(() -> this.close());
@VisibleForTesting
bool shutdownHookAdded;
```
...
```
if (builder.closeOnShutdown) {
    shutdownHookAdded = builder.closeOnShutdown;
    Runtime.getRuntime().addShutdownHook(shutdownHook);
}
```
You can test if shutdownHookAdded is set to true. Since entire test runs as a process it is hard to test if Runtime hook was added in this specific test